presto-orc/src/main/java/io/prestosql/orc/reader/TimestampColumnReader.java

```diff
@@ -512,6 +512,13 @@ private void readTimestampNanos(int i, long[] microsValues, int[] picosFractionV
             picosFraction = toIntExact(nanos * PICOSECONDS_PER_NANOSECOND);
         }
 
+        if (!isFileUtc()) {
```

Comment:
This should go before `Support for variable precision timestamps in Hive connector` commit (it's currently after).

Also, is this change indivudally testable without your other changes?
if not, you can just squash them.